### PR TITLE
i1588: Refactor bb8 to get configs from external source

### PR DESCRIPTION
**Pull request by MartinEden**
In the next stage of this work, inside of copying the configs in we will build them into the Docker image.

From now on, we should be running bb8 from the montagu-bb8 repo, where it is a submodule.

I do three things here:

* Move to getting config from a user-specified file (so when running from within `montagu-bb8` it can be specified as `../config.json`).
* Allow user to choose targets from a menu, rather than machine specific configs
* And in order to support this new behaviour, setting up the configs is now a separate `setup_config.py` script that you run before `setup.sh` (although you will be prompted by setup.sh if you do things in the wrong order).

### CODE REVIEW

File: targets.py [hunk 1]
```diff
@@ -2,34 +2,44 @@
...
     def id(self):
         return "Named volume: " + self.name
 
+    @property
+    def mount_id(self):
+        return self.volume
+
     def _volume_exists(self):
         text = run(
             ["docker", "volume", "ls", "-q"],
```
Comment (richfitz) [inline targets.py:37]: can you use the docker python client instead?

```python
self.volume in [x.name for x in self.docker.volumes.list()]
```

and in the object initialisation either pass in a docker client or construct with `docker.client.from_env()`)

File: targets.py [hunk 2]
```diff
@@ -2,34 +2,44 @@
...
         names = text.split('\n')
-        return self.name in names
+        return self.volume in names
 
     def before_restore(self):
         if not self._volume_exists():
-            print("Creating docker volume with name '{}'".format(self.name))
-            run(["docker", "volume", "create", "--name", self.name], stdout=PIPE)
+            print("Creating docker volume with name '{}'".format(self.volume))
+            run(["docker", "volume", "create", "--name", self.volume], stdout=PIPE)
```
Comment (richfitz) [inline targets.py]: similarly, use

```python
self.docker.volumes.create(self.volume)
```

File: targets.py (hunk 1 above)
Comment (richfitz) [reply targets.py:37]: (or just use `self.docker.volumes.get(self.volume)` with a `except docker.errors.NotFound:` clause)

**Review by richfitz — CHANGES_REQUESTED** [review]
Can you move to the python docker client - we already depend on it

File: targets.py (hunk 1 above)
Comment (MartinEden) [reply targets.py:37]: Opened https://vimc.myjetbrains.com/youtrack/issue/VIMC-1590 for this

File: targets.py (hunk 2 above)
Comment (MartinEden) [reply targets.py]: Opened https://vimc.myjetbrains.com/youtrack/issue/VIMC-1590 for this

**Review by hillalex — APPROVED** [review]
Cool, I like this way of handling the config, feels cleaner

**Pull request merged**